src/DominantJuice.sol

```diff
@@ -0,0 +1,396 @@
+
+        // Calculates the refund bonus, based on the time and size of the pledge
+        uint256 pledgerRefundBonus = calculateRefundBonus(redeemer);
+
+        // Sanity Check
+        if (address(this).balance < pledgerRefundBonus) revert InsufficientFunds();
+
+        // This redeeem function is now locked for the current redeeming address.
+        hasBeenRefunded[redeemer] = true;
+
+        // Sending early pledger refund bonus.
+        (bool sendSuccess,) = redeemer.call{value: pledgerRefundBonus}("");
+        require(sendSuccess, "Failed to send refund bonus.");
+        emit CycleRefundBonusWithdrawal(redeemer, pledgerRefundBonus);
+    }
+
+    /**
+     * @dev This function is callable only after cycle expiry if goal is met.
+     * @param receivingAddress Contract Owner must call this function, but they can
+     * input another address to receive funds if desired.
+     */
+    function creatorWithdraw(address payable receivingAddress, uint256 amount) external payable onlyOwner {
+        require(
+            isCycleExpired == true && isTargetMet == true, "Cycle must be expired and successful to call this function."
+        );
+        if (amount > address(this).balance) revert InsufficientFunds();
+
+        (bool success,) = receivingAddress.call{value: amount}("");
+        require(success, "Failed to withdraw cycle funds.");
+
+        emit OwnerWithdrawal(receivingAddress, amount);
+
+        if (address(this).balance == 0) {
+            hasCreatorWithdrawnAllFunds = true;
+        }
+    }
+
+    /// Getters and Helpers
+
+    // Due to decimnal precision and calculating the power of floating-point numbers, we use PRBMath for this helper function.
+    function calculateRefundBonus(address _pledger) public returns (uint256) {
+        require(pledgedAmount[_pledger] > 0, "Address is not a pledger.");
+
+        uint256 numPledgers = pledgers.length;
+        UD60x18 individualRefundBonus;
+        UD60x18 pledgeAmount_orderLoop;
+        UD60x18 pledgeAmount_amountLoop;
+
+        // Donations and order of pledges are weighted to benefit earlier, larger pledges
+        UD60x18 orderWeight;
+        UD60x18 sumOfOrderWeights;
+        // Memory array of normalized order weights (w-1s)
+        uint256[] memory orderWeightNormalized = new uint[](numPledgers);
+        UD60x18 sumOfAmountsXNormalizedOrderWeights;
+        // uint256 donationWeight;
+        // uint256 donationWeightNormalized;
+        // Represents the combined order and pledge amount weighting.
+        UD60x18 overallPledgeWeight;
+        emit Test("totalAmountPledged", totalAmountPledged);
+        // First, get the sum of pledge order weights to use in the normalizing loop
+        for (uint32 i = 0; i < numPledgers; i++) {
+            // Add 1 to i because rateOfDecay is being raised to the pledge ordereth power
+            sumOfOrderWeights = add(sumOfOrderWeights, rateOfDecay.powu(i + 1));
```

Comment:
a simpler approach would be to take the pledge amount and multiply it by the order weight here. then you get a total amount. normalizing the pledge amounts later against the sum of order weights does nothing mathematically to the pledge amounts except confuse things. the sum of amounts later would just be a bigger number, which you would divide the pledge amounts by anyway

Reply:
I think it would be good for you to take this comment and the previous one to Scott, since he's the man with the formula plan. In the meantime, I'll leave this in while I wait for direction.

Reply:
done